### DISCUSSION TITLE
How to organize large RESTFul application

### DISCUSSION THREAD

**Discussion by textarea** [phalcon/cphalcon#15512] [Q&A]
I have a RESTFul application as documented https://docs.phalcon.io/4.0/en/tutorial-rest

Now I'm getting a lot of hooks, all managed in index.php file and it's growing a lot.

How can I split index.php in subfiles?

**Comment by quasipickle** [comment]
By hooks do you mean:
```php
$app->get(
    '/api/robots',
    function () {
    }
);
```
for example?  If so, they're typically called "routes".

How you organize them generally depends on how large your application is.  The simplest solution would be to just put each route in a separate file, named for the method & URL they handle, ie: `get--api-robots.php`.  Then update your index.php to just `include()` the route file:

```php
<?php

use Phalcon\Mvc\Micro;

$app = new Micro();

include "routes/get--api-robots.php";
include "routes/get--api-robots-search-name.php";
include "routes/get--api-robots-search-id.php";
include "routes/post--api-robots.php";

$app->handle(
    $_SERVER["REQUEST_URI"]
);
```

**Reply by textarea** [reply]
Hi @quasipickle, thank you, I just mean that. Four methods each route grow rapidly.
I hope some more structured solution, for example I don't have auto-completition in routes subdir files.

**Reply by quasipickle** [reply]
I suppose you could build some auto-loading functionality that uses the method and request path to find the file to load.

**Reply by kaioken** [reply]
@textarea I would suggest taking a look at https://github.com/phalcon/rest-api (rest-api example)

**Reply by niden** [reply]
@textarea Have a look at the repository that @kaioken posted and how it is structured. You will find that each action has its own controller, making your code a lot smaller (per action).

I am sure that that or something similar will help your project.
